server/src/test/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResourceTest.java

```diff
@@ -0,0 +1,72 @@
+package keywhiz.service.resources.automation.v2;
+
+import java.util.Collections;
+import java.util.UUID;
+import javax.inject.Inject;
+import keywhiz.KeywhizTestRunner;
+import keywhiz.service.daos.SecretDAO;
+import keywhiz.service.daos.SecretSeriesDAO;
+import org.jooq.DSLContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static keywhiz.jooq.tables.Secrets.SECRETS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(KeywhizTestRunner.class)
+public class BackfillRowHmacResourceTest {
+  @Inject private BackfillRowHmacResource resource;
+  @Inject private SecretDAO.SecretDAOFactory secretDAOFactory;
+  @Inject private SecretSeriesDAO.SecretSeriesDAOFactory secretSeriesDAOFactory;
+  @Inject private DSLContext dslContext;
+
+  private SecretDAO secretDAO;
+  private SecretSeriesDAO secretSeriesDAO;
+
+  @Before
+  public void before() {
+    secretDAO = secretDAOFactory.readwrite();
+    secretSeriesDAO = secretSeriesDAOFactory.readwrite();
+  }
+
+  @Test
+  public void backfillsNonNullSecretHmac() {
```

Comment:
I had another test which attempted to verify that it would also backfill a null HMAC, however, it turns out that that isn't allowed by the database schema.